src/views/lighttable.c

```diff
@@ -1487,14 +1490,67 @@ static gboolean star_key_accel_callback(GtkAccelGroup *accel_group, GObject *acc
   dt_view_t *self = darktable.view_manager->proxy.lighttable.view;
   int num = GPOINTER_TO_INT(data);
   int32_t mouse_over_id;
+  int next_image_rowid = -1;
 
-  mouse_over_id = dt_view_get_image_to_act_on();
+  dt_library_t *lib = (dt_library_t *)self->data;
+  if (lib->using_arrows)
+  {
+    // if using arrows may be the image I'm rating is going to disappear from the collection. So, store where may be we need to jump
+    int imgid_for_offset;
+    sqlite3_stmt *stmt;
 
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+        "SELECT MIN(imgid) FROM selected_images", -1, &stmt,
+        NULL);
+    if(sqlite3_step(stmt) == SQLITE_ROW)
+    {
+      imgid_for_offset = sqlite3_column_int(stmt, 0);
+      if (!imgid_for_offset)
+        // empty selection
+        imgid_for_offset = dt_control_get_mouse_over_id();
+
+      DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+          //"SELECT imgid FROM memory.collected_images", -1, &stmt,
+          "SELECT rowid FROM memory.collected_images where imgid=?1", -1, &stmt,
```

Comment:
You should create a new `sqlite_stmt*` (we normally call it `inner_stmt`) and nor reuse `stmt`. Don't forget to finalize both.